processing/src/main/java/org/apache/druid/query/timeboundary/TimeBoundaryQueryRunnerFactory.java

```diff
@@ -155,7 +156,8 @@ public Iterator<Result<TimeBoundaryResultValue>> make()
               final DateTime minTime;
               final DateTime maxTime;
 
-              if (legacyQuery.getFilter() != null) {
+              Interval queryInterval = legacyQuery.getQuerySegmentSpec().getIntervals().get(0);
```

Comment:
Are we guaranteed that `getIntervals()` will always return a list with at least one element?

There are no javadocs on `QuerySegmentSpec#getIntervals()` so I'm not sure if a future implementation may choose to return an empty intervals.

NOTE to self: Why do we need this change?

Reply:
I've improved the code for checking the containment of the adapter interval. Does it look ok now? Now, it always expects exactly 1 interval as other engines do. But still empty intervals pass in timeBoundary because the broker code doesn't generate query runners incase empty intervals are given.
For null intervals (which seems like a valid case for timeBoundary), the query object itself creates an eternity interval.